### DISCUSSION TITLE
s3 backup in non aws s3 storages

### DISCUSSION THREAD

**Discussion by Citrullin** [longhorn/longhorn#4203] [Q&A] [answered]
I am confused how to backup into non aws s3. I added the s3 endpoint to the endpoint field of longhorn, but I get an error message that https is not supported. The documentation states to use `s3://backupbucket@us-east-1/` for minio. That doesn't seem right.
Can someone tell me where to put the endpoint and what endpoint is expected. The secret config looks straight forward though.
When I only configure it without endpoint, I get the error `backup target URL is empty`.
The secret is already available under the namespace `onghorn-system`, but it also doesn't work without UI config.

**Comment by mantissahz** [comment]
I think you had followed this doc [set-up-aws-s3-backupstore](https://longhorn.io/docs/1.3.0/snapshots-and-backups/backup-and-restore/set-backup-target/#set-up-aws-s3-backupstore) to setup the s3 backup store.
`s3://backupbucket@us-east-1/` is the correct format for longhorn to use the AWS s3 as a backup target.

**Reply by Citrullin** [reply]
> I think you had followed this doc [set-up-aws-s3-backupstore](https://longhorn.io/docs/1.3.0/snapshots-and-backups/backup-and-restore/set-backup-target/#set-up-aws-s3-backupstore) to setup the s3 backup store. `s3://backupbucket@us-east-1/` is the correct format for longhorn to use the AWS s3 as a backup target.

It doesn't matter if this endpoint doesn't exist?

**Reply by Citrullin** [reply]
> I think you had followed this doc [set-up-aws-s3-backupstore](https://longhorn.io/docs/1.3.0/snapshots-and-backups/backup-and-restore/set-backup-target/#set-up-aws-s3-backupstore) to setup the s3 backup store.
> `s3://backupbucket@us-east-1/` is the correct format for longhorn to use the AWS s3 as a backup target.

```
error listing backup volume names: failed to execute: /var/lib/longhorn/engine-binaries/longhornio-longhorn-engine-v1.3.0/longhorn [backup ls --volume-only s3://k8-backups@fr-par/], output failed to list objects with param: { Bucket: "k8-backups", Delimiter: "/", Prefix: "/" } error: AWS Error: NoSuchKey The specified key does not exist. <nil> 404 tx547183728cad4db9a80c5-0062c3714e , stderr, time="2022-07-04T23:01:34Z" level=error msg="Fail to list s3: failed to list objects with param: {\n Bucket: \"k8-backups\",\n Delimiter: \"/\",\n Prefix: \"/\"\n} error: AWS Error: NoSuchKey The specified key does not exist. <nil>\n404 tx547183728cad4db9a80c5-0062c3714e\n" pkg=s3 time="2022-07-04T23:01:34Z" level=error msg="failed to list objects with param: {\n Bucket: \"k8-backups\",\n Delimiter: \"/\",\n Prefix: \"/\"\n} error: AWS Error: NoSuchKey The specified key does not exist. <nil>\n404 tx547183728cad4db9a80c5-0062c3714e\n" , error exit status 1
```
Well, it doesn't work.

**Reply by mantissahz** [reply]
> It doesn't matter if this endpoint doesn't exist?

Yes, your endpoint should exist.

> s3://k8-backups@fr-par/

Could you check this `"<your-bucket-name>/*"` is set for permissions at [set-up-aws-s3-backupstore](https://longhorn.io/docs/1.3.0/snapshots-and-backups/backup-and-restore/set-backup-target/#set-up-aws-s3-backupstore)?

**Reply by mantissahz** [reply]
Another ref: [issues-3333](https://github.com/longhorn/longhorn/issues/3333#issuecomment-981230972) for you.
Is your s3 server minio?

**Comment by c3y1huang** [comment, marked as answer]
Ref https://github.com/longhorn/longhorn/issues/4205#issuecomment-1174616197

**Comment by angelocorreia27** [comment]
I wonder when we can have this functionality. it would be very useful. I've been trying some solutions like: https://github.com/longhorn/longhorn/issues/4205#issuecomment-1429149107
But it just doesn't work.

**Comment by spuxx-dev** [comment]
Hetzner recently released an AWS-S3-compatible object storage. The actual endpoints look like this: `https://name.region.your-objectstorage.com`. For example, a bucket named `foo` at region `nbg1` would be: https://foo.nbg1.your-objectstorage.com/ (clickable)

I tried a couple of conbinations to set this up with longhorn but couldnt get it to work. @mantissahz would you happen to know how I'd need to set up the URLs in longhorn?

/ Edit: I got it to work. See here: https://github.com/longhorn/longhorn/issues/4205#issuecomment-2558091327
